contracts/gauge/Gauge.sol

```diff
@@ -0,0 +1,217 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.6.7; 
+
+import "../libraries/SafeMath.sol";
+import "../libraries/SafeERC20.sol";
+import "../libraries/ProtocolGovernance.sol";
+import "../libraries/Math.sol";
+import "../libraries/reentrancy-guard.sol";
+
+
+contract Gauge is ProtocolGovernance, ReentrancyGuard {
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+
+    event Staked(address indexed user, uint256 amount);
+    event Withdrawn(address indexed user, uint256 amount);
+    event RewardPaid(address indexed user, uint256 reward, uint256 tokenIndex);
+    event RewardAdded(uint256 reward);
+
+    IERC20 public constant AXIAL = IERC20(0xcF8419A615c57511807236751c0AF38Db4ba3351); 
+    IERC20 public constant TREASURY = IERC20(0x4980AD7cCB304f7d3c5053Aa1131eD1EDaf48809);
+
+    IERC20 public XAXIAL = IERC20(0xcF8419A615c57511807236751c0AF38Db4ba3351); // TODO: Fix this
+    IERC20 public immutable TOKEN = IERC20(0xcF8419A615c57511807236751c0AF38Db4ba3351); // TODO: Fix this
+
+    address[] rewardTokens; 
+    address public DISTRIBUTION;
+    uint256 public constant DURATION = 7 days;
+
+    uint256 public periodFinish = 0;
+    mapping(address => uint256) public rewardRates;
+    mapping(address => uint256) public rewardPerTokenStored;
+
+    uint256 public lastUpdateTime;
+
+    mapping(address => mapping(address => uint256)) public userRewardPerTokenPaid;
+    mapping(address => mapping(address => uint256)) public rewards;
+
+    uint256 private _totalSupply;
```

Comment:
I believe our convention of using the _ prefix on member variables implies that something represents the balance of the non-prefixed member variable.  i.e. Token points to the address of our token whereas _token represents the balance of that token.